File: 485.max-consecutive-ones.py
#
# @lc app=leetcode id=485 lang=python3
#
# [485] Max Consecutive Ones
#

# @lc code=start
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:

        max_consecutives = 0
        current_consecutives = 0

        for val in nums:
            if val == 0:
                max_consecutives = max(max_consecutives, current_consecutives)
                current_consecutives = 0
            else:
                current_consecutives += 1
        
        return max(max_consecutives, current_consecutives)
        
# @lc code=end

File: 485.max-consecutive-ones.java
/*
 * @lc app=leetcode id=485 lang=java
 *
 * [485] Max Consecutive Ones
 */

// @lc code=start
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxConsecutives = 0;
        int currentConsecutives = 0;

        for (int val : nums) {
            if (val == 0) {
                maxConsecutives = Math.max(maxConsecutives, currentConsecutives);
                currentConsecutives = 0;
            } else {
                currentConsecutives++;
            }
        }

        return Math.max(currentConsecutives, maxConsecutives);
    }
}
// @lc code=end

